app/models/profile.rb

```diff
@@ -60,7 +60,7 @@ def results(host)
               ON rules.id = profile_rules.rule_id
               WHERE profile_rules.profile_id = ?)
        ) rule_results WHERE RANK = 1', host.id, id]
-      )
+      ).select { |rr| %w(pass fail notapplicable error unknown).include?(rr.result) }
```

Comment:
Does this work?

```suggestion
      ).where(result: %w(pass fail notapplicable error unknown))
```

Reply:
Using `.where` might not, `find_by_sql` seems to return a plain `Array`, but if the SQL query could/would be transformed into a Arel query it should work.

Reply:
Oh, then it should also be possible to use it with the `:selected` scope, @akofink proposed.

Reply:
`NoMethodError (undefined method `where' for #<Array:0x000056298f41c670>):`

Then again I can add just the where to the query directly